### PR TITLE
[DRAFT] OpenSWATH for Metabolomics

### PR DESCRIPTION
**Pull request by oliveralka**
Draft on how to use the DIAMetAlyzer workflow. 

The current workflow depicts the whole analysis of the benchmarking data - this is probably too much for a short tutorial session. 

It will help to understand the workflow, reanalyse the data and probably use it with own data. 

Let me know what you think!

### CODE REVIEW

**Comment by oliveralka** [comment]
Link to PDF version (20200514):
https://drive.google.com/open?id=1Snar3Vhpto02TisAV08OB0CmAIDdfcKW

**Comment by oliveralka** [comment]
[handout_add_DIAMetAlyzer_20200723.pdf](https://github.com/OpenMS/Tutorials/files/4964401/handout_add_DIAMetAlyzer_20200723.pdf)
@jpfeuffer Please let me know what you think when you have time. If everything is alright we can merge.

**Comment by jpfeuffer** [comment]
So, 7.2. you could make some paragraphs.
In the workflow, you could merge all the Input variables into a Wrapped Metanode (or whatever this is called these days haha). Then you can make a little Wizard for the user so they can enter those parameters with a little help/description.

**Comment by oliveralka** [comment]
Ok, that is pretty neat idea - I will give it a try!

**Comment by oliveralka** [comment]
Ok, I added the paragraphs. For the configuration, I switched to a component with configuration nodes, which will open a configuration menu directly. 

I tested the interactive view using widgets, but I found that if everything else is "normally" configurated (e.g. input files) the opening the browser to configure four additional variables is not very intuitive.

**Comment by jpfeuffer** [comment]
Yes, for input files you would another input variable and pass it and for multiple files this also not straightforward, so I usually don't use it for files as well.
It was a suggestion because it was four variable nodes and one might save work. Yes, it is different from the usual configuration procedure. Complete KNIME newbies might struggle. We have a small chapter on meta nodes I think.
Concerning paragraphs: looks good. I did not expect section numbers, just paragraphs, but this is fine too of course.

**Comment by oliveralka** [comment]
@jpfeuffer Thanks for taking a look - switched to paragraphs - looks better! Should be ready to merge now!

File: Handout/openswathmetabo.tex [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+\end{tabular*}
+\end{center}
+
+\noindent\KNIMENODE{\textbf{AccurateMassSearch}}:
+\begin{center}
+\begin{tabular*}{\textwidth}{ p{5.5cm}|p{10.5cm} }
+\textbf{parameter} & \textbf{explanation} \\ \hline
+\textit{mass\_error\_value} & Tolerance allowed for accurate mass search. \\
+\textit{ionization\_mode} &  Positive or negative ionization mode. \\
+\textit{mass\_error\_value} & Tolerance allowed for accurate mass search. \\
```
Comment (jpfeuffer) [inline Handout/openswathmetabo.tex:152]: duplication

File: Handout/openswathmetabo.tex [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+\noindent\KNIMENODE{\textbf{OpenSWATH}}:
+\begin{center}
+\begin{tabular*}{\textwidth}{ p{5.5cm}|p{10.5cm} }
+\textbf{parameter} & \textbf{explanation} \\ \hline
+\textit{rt\_extraction\_window} & Extract x seconds around this value. \\
+\textit{rt\_normalization\_factor} &  Please use the range of your gradient e.g. 950 seconds. \\
+\end{tabular*}
+\end{center} 
+
+\noindent If you are analysing a lot of big DIA mzML files ~3-20GB per File, it makes sense to  change how OpenSWATH processes the spectra.
```
Comment (jpfeuffer) [inline Handout/openswathmetabo.tex:181]: \approx

File: Handout/openswathmetabo.tex [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+\begin{figure}[!h]
+  \centering
+  \includegraphics[width=0.95\textwidth]{graphics/openswathmetabo/oswm_example_wf.png}
+  \caption{Example workflow for the usage of the DIAMetAlyzer Pipeline in KNIME}
+  \label{fig:oswm_example_wf}
+\end{figure}
+
+\subsection{Run the Workflow}
+These steps need to be followed to run the workflow successfully
```
Comment (jpfeuffer) [inline Handout/openswathmetabo.tex:105]: ":"?

File: Handout/openswathmetabo.tex [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+\noindent The example data can be found here:
+\url{https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Tutorials/Data/DIAMetAlyzer/}
+
+\subsection{Example Workflow}
+Example workflow for the usage of the DIAMetAlyzer Pipeline in KNIME (see Fig. \ref{fig:oswm_example_wf}). Inputs are the SWATH-MS data in profile mode (.mzML), a path for saving the new target-decoy assay library, the SIRIUS 4.0.1 executable, the DDA data (.mzML), custom libraries and adducts for AccurateMassSearch, the min/max fragment mass-to-charge to be able to restrict the mass of the transitions and the path to the PyProphet executable. The DDA is used for feature detection, adduct grouping, accurate mass search and forwarded to the AssayGeneratorMetabo. The constructed target library is used by the decoy generation node, which calls various python scripts to parse, reformat the input, call Passatutto and export the target-decoy assay library. The target-decoy assay library is processed with the SWATH-MS data in OpenSWATH. The results are used by PyProphet for scoring and output a list of metabolites with their respective q-value and quantitative information
```
Comment (jpfeuffer) [inline Handout/openswathmetabo.tex:95]: Missing punctuation at end?

File: Handout/openswathmetabo.tex [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+\end{tabular*}
+\end{center} 
+
+\noindent In the workflow pyprophet is called after OpenSWATH, it merges the result files, which allows to get enough data for the model training. 
+
+\begin{lstlisting}
+pyprophet merge  --template path_to_target-decoy_assay_library.pqp --out merged.osw  ./*.osw
+\end{lstlisting}
+
+\noindent Afterwards, the results are scored using the MS1 and MS2 leves and filter for metabolomics scores, which have a low correlation.
```
Comment (jpfeuffer) [inline Handout/openswathmetabo.tex:197]: levels

File: Handout/openswathmetabo.tex (hunk 2 above)
Comment (jpfeuffer) [reply Handout/openswathmetabo.tex:181]: double space "to  change"

**Pull request merged**